src/armlsd.py

```diff
@@ -60,6 +60,7 @@ def ARMLSD(argflag, spect, fitsdict, reuseMaster=False):
         slf = sciexp[sc]
         scidx = slf._idx_sci[0]
         msgs.info("Reducing file {0:s}, target {1:s}".format(fitsdict['filename'][scidx], slf._target_name))
+        msgs.sciexp = slf  # For QA writing on exit, if nothing else.  Could write Masters too
```

Comment:
I'm happy with your approach, but it's probably worthwhile having a brief discussion about this.

If a bug is encountered (and not an error message), then the QA will not be dumped. The only way this can be done is to throw a general try/except clause around ARMLSD, which I hesitate to do (because there's already a try/except clause enclosing the call to PYPIT).

It seems we can just include `slf.sciexp._qa.savefig()` whenever we want a page written to be written to file. See: [this link](http://matplotlib.org/examples/pylab_examples/multipage_pdf.html)

I've not tried this, so I'm not sure if it will work.

Another option might be to remove the try/except around PYPIT, and instead put it within ARMLSD, specifically:

```
def ARMLSD(...):
    try:
        reduce data
    except:
        slf.sciexp._qa.close()
        then, print traceback, as in the __init__ function in pypit.py
```

which, again, seems like a dirty answer to our problem.